packages/camera/camera_android_camerax/lib/src/image_proxy.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart' show BinaryMessenger;
+import 'package:meta/meta.dart' show protected;
+
+import 'android_camera_camerax_flutter_api_impls.dart';
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+import 'plane_proxy.dart';
+
+/// Representation of a single complete image buffer.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/ImageProxy.
+class ImageProxy extends JavaObject {
+  /// Constructs a [ImageProxy] that is not automatically attached to a native object.
+  ImageProxy.detached(
+      {BinaryMessenger? binaryMessenger, InstanceManager? instanceManager})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = _ImageProxyHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  late final _ImageProxyHostApiImpl _api;
+
+  /// Returns the list of color planes of image data.
+  Future<List<PlaneProxy>> getPlanes() => _api.getPlanesFromInstances(this);
```

Comment:
I should convert these getters to class fields.

Reply:
See https://github.com/flutter/packages/pull/3454#discussion_r1174443624

Reply:
As I stated in another comment, I think everything besides this one could be a fields since they are not expensive to pass. Can be done in another PR though.